clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -102,12 +123,17 @@ public boolean equals(Object obj) {
             return false;
         if (port != other.port)
             return false;
+        if (rack == null) {
+            if (other.rack != null)
+                return false;
+        } else if (!rack.equals(other.rack))
+            return false;
         return true;
     }
 
     @Override
     public String toString() {
-        return host + ":" + port + " (id: " + idString + ")";
+        return host + ":" + port + " (id: " + idString + " rack: " + rack + ")";
```

Comment:
We are using this `toString` in a number of log statement, not sure the rack information is that relevant (the fact that `toString` is used for both debug and more user-facing string is a pain). @hachikuji, you cleaned up the logs, what do you think?